hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -625,6 +625,9 @@ private static void processRestoreMetadata(HoodieActiveTimeline metadataTableTim
 
   /**
    * Convert rollback action metadata to metadata table records.
+   * <p>
+   * We only need to handle FILES partition here as HUDI rollbacks on MOR table may end up adding a new log file. All other partitions
+   * are handled by actual rollback of the deltacommit which added records to those partitions.
```

Comment:
why we do this. Triggering rollback for the MDT will actually add rollback blocks to all partitions in MDT right. So, no additional handling should be required in my understanding. 
or am I missing something.

Reply:
This is for the case of MOR table restores/rollbacks which may add additional files (log files).